src/rpc.ts

```diff
@@ -45,16 +48,54 @@ router.post('/', async (req, res) => {
       return rpcError(res, 500, 'Unable to authenticate your verification link', id);
     } else if (method === 'snapshot.unsubscribe') {
       if (verifyUnsubscribe(params.email, params.signature)) {
-        await unsubscribe(params.email);
+        await unsubscribe(params.email, params.subscriptions);
```

Comment:
`verifyUnsubscribe` is here to prevent that, you can not pass any email, else the signature will fail

Reply:
with the current implementation, no one can unsubscribe, because we assume that the signer is the one whose private key is defined in `process.env.WALLET_PRIVATE_KEY`. That's why I proposed avoiding `signer || wallet.address`. 
Looking at `verifyUnsubscribe` we can assume that it receives only email and signature, from the signature you can get the signer, but you cannot compare it with the signer, since we don't pass it to `verifyUnsubscribe`.

Reply:
backend sign(email) then send signature to user
user return the signature, backend only verify(email), and check that signer is himself

Reply:
Thank you I got the case.